frappe/public/js/frappe/list/list_view.js

```diff
@@ -938,27 +908,174 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 			};
 		};
 
+		const bulk_delete = () => {
+			return {
+				label: __('Delete'),
+				action: () => {
+					const docnames = this.get_checked_items(true);
+					if (docnames.length > 0) {
+						frappe.confirm(__('Delete {0} items permanently?', [docnames.length]),
+							() => {
+								frappe.call({
+									method: 'frappe.desk.reportview.delete_items',
+									freeze: true,
+									args: {
+										items: docnames,
+										doctype: this.doctype
+									}
+								}).then((r) => {
+									let failed = r.message;
+									if (!failed) failed = [];
+
+									if (failed.length && !r._server_messages) {
+										frappe.throw(__('Cannot delete {0}', [failed.map(f => f.bold()).join(', ')]));
+									}
+									if (failed.length < docnames.length) {
+										frappe.utils.play_sound('delete');
+										this.refresh(true);
+									}
+								});
+							}
+						);
+					} else {
+						frappe.msgprint(__('Select atleast 1 record to delete'));
+					}
+				},
+				standard: true,
+			}
+		}
+
+		const bulk_cancel = () => {
+			return {
+				label: __('Cancel'),
+				action: () => {
+					const items = this.get_checked_items(true);
+					if (items.length > 0) {
+						frappe.confirm(__('Cancel {0} documents?', [items.length]),
+							() => {
+								frappe
+									.call({
+										method: 'frappe.desk.doctype.bulk_update.bulk_update.cancel_items',
+										args: {
+											doctype: doctype,
+											items: items
+										},
+										callback: this.refresh()
+									});
+							});
+					}
+				},
+				standard: true
+			}
+		};
+
+		const bulk_submit = () => {
+			return {
+				label: __('Submit'),
+				action: () => {
+					const items = this.get_checked_items(true);
+					if (items.length > 0) {
+						frappe.confirm(__('Submit {0} documents?', [items.length]),
+							() => {
+								frappe
+									.call({
+										method: 'frappe.desk.doctype.bulk_update.bulk_update.submit_items',
+										args: {
+											doctype: doctype,
+											items: items
+										},
+										callback: this.refresh()
+									});
+							});
+					}
+				},
+				standard: true
+			}
+		};
+
+		const bulk_edit = () => {
+			return {
+				label: __('Edit'),
+				action: () => {
+					let fields_to_skip = [];
+					let field_options = [];
+					let field_mappings = {};
+
+					frappe.meta.get_docfields(doctype).forEach(field_doc => {
+						if (is_field_editable(field_doc)) {
+							field_options.push(field_doc.label);
+							field_mappings[field_doc.label] = Object.assign({}, field_doc);
+						}
+					});
+
+					if(!field_options.length) {
+						frappe.msgprint('No editable fields found.');
+					}
+
+					const dialog = new frappe.ui.Dialog({
+						title: __("Edit"),
+						fields: [{
+							"fieldtype": "Select", "options": field_options,
```

Comment:
`fieldtype: 'Autocomplete'`

Reply:
Skipped this one (as it will take validation on field names for not much gain)
instead sorted the field options so that it will easier for the user to select the field.